src/nvim/api/vim.c

```diff
@@ -663,6 +663,65 @@ Dictionary nvim_get_color_map(void)
   return colors;
 }
 
+/// Get the mappings for the corresponding scope and mode
+///
+/// @param mode
+/// @param scope
+Array nvim_get_keymap(String mode, String scope)
+{
+  Array mappings = ARRAY_DICT_INIT;
+  mapblock_T *current_maphash;
+
+  // Convert the string mode to the integer mode
+  // that is stored within each mapblock
+  int int_mode = get_map_mode((char_u **)(&(mode.data)), 0);
+
+  int i = 0;
+  for (i = 0; i < MAX_MAPHASH; i++) {
+    // Get the correct maphash based on global or local hashes
+    current_maphash = get_maphash(i, scope.data);
+
+    if (current_maphash == NULL) {
+      continue;
+    }
+
+    while (current_maphash) {
+      // Check for correct mode
+      if ((int_mode & current_maphash->m_mode) != 0) {
+        Object maparg_dict = OBJECT_INIT;
+
+        Array args = ARRAY_DICT_INIT;
+        // Lhs
+        ADD(args, STRING_OBJ(cstr_to_string((char *)current_maphash->m_keys)));
+        // Mode
+        ADD(
+            args,
+            STRING_OBJ(
+                cstr_to_string(map_mode_to_chars(current_maphash->m_mode))));
+        // Not abbr
+        ADD(args, INTEGER_OBJ(0));
+        // Dictionary result
+        ADD(args, INTEGER_OBJ(1));
+
+        Error err = ERROR_INIT;
+
+        maparg_dict = nvim_call_function(cstr_to_string("maparg"), args, &err);
+
+        if (err.set) {
+          ADD(mappings, STRING_OBJ(cstr_to_string("this one didn't work")));
```

Comment:
I would use `assert(false)` here: missing description for an existing map means that something is broken, better raise the possibility it will be reported.

Though I do not like the idea of using `maparg` function in first place: you just need to copy and refactor ~15 mostly trivial lines from `get_maparg`, this is not enough to bother with reusing code.